sebs/faas/nosql.py

```diff
@@ -0,0 +1,122 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import Dict, Optional, Tuple
+
+from sebs.faas.config import Resources
+from sebs.cache import Cache
+from sebs.utils import LoggingBase
+
+
+class NoSQLStorage(ABC, LoggingBase):
+    @staticmethod
+    @abstractmethod
+    def deployment_name() -> str:
+        pass
+
+    @property
+    def cache_client(self) -> Cache:
+        return self._cache_client
+
+    @property
+    def region(self):
+        return self._region
+
+    def __init__(self, region: str, cache_client: Cache, resources: Resources):
+        super().__init__()
+        self._cache_client = cache_client
+        self._cached = False
+        self._region = region
+        self._cloud_resources = resources
+
+        # Map benchmark -> orig_name -> table_name
+        # self._tables: Dict[str, Dict[str, str]] = defaultdict(dict)
+
+    @abstractmethod
+    def get_tables(self, benchmark: str) -> Dict[str, str]:
+        pass
+
+    @abstractmethod
+    def _get_table_name(self, benchmark: str, table: str) -> Optional[str]:
+        pass
+
+    @abstractmethod
+    def retrieve_cache(self, benchmark: str) -> bool:
+        pass
+
+    @abstractmethod
+    def update_cache(self, benchmark: str):
+        pass
+
+    def envs(self) -> dict:
+        return {}
+
+    """
+        Each table name follow this pattern:
+        sebs-benchmarks-{resource_id}-{benchmark-name}-{table-name}
+
+        Each implementation should do the following
+        (1) Retrieve cached data
+        (2) Create missing table that do not exist
+        (3) Update cached data if anything new was created
+    """
+
+    def create_benchmark_tables(
```

Comment:
The singular form (`create_benchmark_table`) would probably be more appropriate.

I am even wondering whether we could only have `create_table` as a method and skip `create_benchmark_table` altogether -- or is there a reason why this is done like this?

Reply:
The main idea was to force a single behavior on all systems: check cache, create a name, and create a table if necessary.